templates/vue/cypress/integration/tyde.spec.js

```diff
@@ -0,0 +1,260 @@
+    "https://lh3.googleusercontent.com/proxy/EfGXb_q2dzJt5VBgQuXuNjJGwLWO0tlSdVz5gTalgRNzkiso3HtplzRhGOpyYsaS4-4kZuda4SUewK0UcnhJPfYmuDzIp1Ohyg"
+
+  const earnedCockpitIcon =
+    "https://www.clipartmax.com/png/middle/29-292956_world-globe-clip-art-clipart-2-wikiclipart-wikiclipart-transparent-background-earth-clipart.png"
+
+  const hoverCockpitIcon =
+    "https://www.clipartmax.com/png/middle/29-292956_world-globe-clip-art-clipart-2-wikiclipart-wikiclipart-transparent-background-earth-clipart.png"
+
+  describe("Module Creation", () => {
+    it("Should allow creation of a Module from the root node", () => {
+      cy.fixture("one-node.json").as("oneNode")
+      cy.setup("@oneNode")
+      cy.getSelectedNode().then(node => {
+        cy.openModal("add", node.id)
+        cy.getByTestId(`node-title`).type("Module 1")
+        cy.getByTestId(`node-media-type`).select("text")
+        cy.getByTestId(`tyde-node-type`).select("Module")
+        cy.getEditable(`node-text-content`).type("This is the main module.")
+        cy.contains(/spaceship part/i).click()
+        cy.getByTestId(`tyde-planet-not-earned-icon`).type(notEarnedIcon)
+        cy.getByTestId(`tyde-planet-earned-icon`).type(earnedIcon)
+        cy.submitModal()
+        cy.getBySrc(earnedIcon).should("exist")
+        cy.getBySrc(notEarnedIcon).should("not.exist")
+      })
+    })
+
+    it("Should allow creation of a Stage node from a module", () => {
+      cy.fixture("tyde/one-module.json").as("oneModule")
+      cy.setup("@oneModule")
+      cy.getNodeByTitle("Module 1").then(node => {
+        cy.openModal("add", node.id)
+        cy.getByTestId(`node-title`).type("Stage 1")
+        cy.getByTestId(`node-media-type`).select("text")
+        cy.getByTestId(`tyde-node-type`).select("Stage")
+        cy.getEditable(`node-text-content`).type(
+          "This is the first stage for module 1."
+        )
+        cy.submitModal()
+        cy.getBySrc(earnedIcon).should("exist")
+        cy.getBySrc(notEarnedIcon).should("not.exist")
+      })
+    })
+
+    it("Should allow creation of a Question Set from a stage", () => {
+      cy.fixture("tyde/one-stage.json").as("oneStage")
+      cy.setup("@oneStage")
+      cy.getNodeByTitle("Stage 1").then(node => {
+        cy.openModal("add", node.id)
+        cy.getByTestId(`node-title`).type("Question Set 1")
+        cy.getByTestId(`node-media-type`).select("text")
+        cy.getByTestId(`tyde-node-type`).select("Question set")
+        cy.getEditable(`node-text-content`).type(
+          "This is the first question set for module 1 stage 1."
+        )
+        cy.submitModal()
+        cy.getBySrc(earnedIcon).should("not.exist")
+        cy.getBySrc(notEarnedIcon).should("exist")
+      })
+    })
+
+    it("Should allow creation of a regular node (subtopic) from a question set", () => {
+      cy.fixture("tyde/one-question-set.json").as("oneQuestionSet")
+      cy.setup("@oneQuestionSet")
+      cy.getSelectedNode().then(node => {
```

Comment:
Submitted a bug report for this but I think there's an issue with getSelectedNode since it seems to be selecting Root, instead of the question set.
![image](https://user-images.githubusercontent.com/45089842/102920212-276cc400-443f-11eb-851b-11226341d2a7.png)

Reply:
Oh good catch! Super easy to get false positives with cypress. I can actually just convert this to get the node by title.